### DISCUSSION TITLE
Make Select field compatible with hasOneThrough

### DISCUSSION THREAD

**Discussion by plantjes** [filamentphp/filament#10365] [Ideas]
Is it possible to make the select field compatible with hasOneTrough?

I have this field in my User Resource, but only the active (chosen) value is visible and not all other schools.
```php
Select::make('school')
	  ->searchable()
	  ->relationship(name: 'school', titleAttribute: 'name' )
	  ->label(__('fields.school')),
``` 

In my User model I have this:

```php
public function school(): hasOneThrough
{
	return $this->hasOneThrough(School::class, SchoolUser::class, 'user_id', 'id', 'id', 'school_id' );
}
``` 

If I save it like this (with only one visible active option) I get this error:

Call to undefined method Illuminate\Database\Eloquent\Relations\HasOneThrough::associate()

I know that it works with the relationship manager, but that is more designed for BelongToMany relationships. 
And I don't want a belongsTo model because my users are also admin or other types of users and that is why i'm using hasOneThrough with a pivot table. So I can keep my users table clean without a school_id or a school_admin_id etc.

**Comment by jimiero** [comment]
@plantjes have you ever found a solution for this ?

**Reply by plantjes** [reply]
I have changed my relation structure and now I'm using belongsToMany

**Reply by coulterpeterson** [reply]
@plantjes would you mind providing an example of how you switched over to belongsToMany? 

I have the exact same scenario where I have User -> Settings -> SettingsLanguagesPivot -> LanguagesTable where all the relationships are defined, but populating a select using the user's 'hasManyThrough' for languages via settings is throwing the same error. Appreciate any help.
